app/controllers/staff/extending_organisations_controller.rb

```diff
@@ -23,6 +23,19 @@ def update
 
   private
 
+  def set_implementing_organisation
+    extending_organisation = Organisation.find(extending_organisation_id)
+    implementing_organisation = ImplementingOrganisation.new(name: extending_organisation.name,
+                                                             organisation_type: extending_organisation.organisation_type,
+                                                             reference: extending_organisation.iati_reference,
+                                                             activity_id: activity_id)
+    if implementing_organisation.valid?
+      implementing_organisation.save
+    else
+      flash[:notice] = I18n.t("activerecord.errors.models.activity.general")
```

Comment:
I haven't added a test for this unhappy path as I'm not sure we "should" ever get here, as the extending organisation should exist and have all its attributes. If we do need a test, I'm not sure how to spec it out 🤔

Reply:
I think the test for this being hard is pointing out that we perhaps don't need the code we cannot test? As setting the implementing organisation is not a user facing action, I am not sure we need the if/else and message, maybe we just call `save!` if something is wrong we get an exception and in this case that feels right, if one of the extending_orgs is not valid something has gone wrong somewhere?

What do you think?

Reply:
I think you're right and I'm being *too* cautious!